src/app/shared/navigation-focus/navigation-focus.spec.ts

```diff
@@ -0,0 +1,128 @@
+import {Component, NgModule, NgZone} from '@angular/core';
+import {ComponentFixture, fakeAsync, TestBed, tick} from '@angular/core/testing';
+import {Router} from '@angular/router';
+import {RouterTestingModule} from '@angular/router/testing';
+import {NavigationFocusService} from './navigation-focus.service';
+import {NavigationFocusModule} from './navigation-focus';
+
+describe('Navigation focus service', () => {
+  let navigationFocusService: NavigationFocusService;
+  let router: Router;
+  let zone: NgZone;
+  let fixture: ComponentFixture<NavigationFocusTest>;
+
+  const navigate = (url: string) => {
+    zone.run(() => router.navigateByUrl(url));
+    tick(100);
+  };
+
+  beforeEach(() => {
+      TestBed.configureTestingModule({
+        imports: [RouterTestingModule.withRoutes([
+          {path: '', component: RouteTest},
+          {path: 'cdk', component: RouteTest},
+          {path: 'guides', component: RouteTest}
+        ]),
+        NavigationFocusModule],
+        providers: [NavigationFocusService],
+        declarations: [NavigationFocusTest, RouteTest],
+      });
+      fixture = TestBed.createComponent(NavigationFocusTest);
+    }
+  );
+
+  beforeEach(() => {
+    zone = TestBed.inject(NgZone);
+    router = TestBed.inject(Router);
+    navigationFocusService = TestBed.inject(NavigationFocusService);
+  });
+
+  it('should set skip link href', () => {
+    const target1 = fixture.nativeElement.querySelector('#target1');
+    const target2 = fixture.nativeElement.querySelector('#target2');
+
+    navigationFocusService.requestSkipLinkFocus(target1);
+    navigationFocusService.requestSkipLinkFocus(target2);
+
+    expect(navigationFocusService.getSkipLinkHref()).toEqual('/#target2');
+
+    navigationFocusService.relinquishSkipLinkFocus(target2);
+
+    expect(navigationFocusService.getSkipLinkHref()).toEqual('/#target1');
+  });
+
+  it('should set id for skip link target without id', () => {
+    const skipLinkTarget = fixture.nativeElement.querySelector('.no-id');
+
+    navigationFocusService.requestSkipLinkFocus(skipLinkTarget);
+
+    expect(navigationFocusService.getSkipLinkHref()).toMatch(`\#skip-link-target-[0-9]*`);
```

Comment:
I don't think you have to escape `#` in a string, or was that supposed to be a forward slash?